clients/drcachesim/simulator/tlb.h

```diff
@@ -46,21 +49,67 @@ namespace drmemtrace {
 
 class tlb_t : public caching_device_t {
 public:
+    tlb_t(const std::string &name = "tlb");
+
     void
     request(const memref_t &memref) override;
 
-    // TODO i#4816: The addition of the pid as a lookup parameter beyond just the tag
-    // needs to be imposed on the parent methods invalidate(), contains_tag(), and
-    // propagate_eviction() by overriding them.
+    // TODO i#4816: The addition of the pid as a lookup parameter beyond just the
+    // tag needs to be imposed on the parent methods invalidate(), contains_tag(),
+    // and propagate_eviction() by overriding them.
+
+    // Returns the replacement policy used by this TLB.
+    virtual std::string
+    get_replace_policy() const override = 0;
 
 protected:
     void
     init_blocks() override;
-
     // Optimization: remember last pid in addition to last tag
     memref_pid_t last_pid_;
 };
 
+/**
+ * TLB with least frequently used replacement policy.
+ *
+ * LFU has us hold a counter per way in the TLB entry.
+ * The counter is incremented when the way is accessed.
+ * When evicting, we choose the way with the least frequent access count.
+ */
+class tlb_lfu_t : public tlb_t {
+public:
+    tlb_lfu_t(const std::string &name = "tlb_lfu");
+
+    std::string
+    get_replace_policy() const override;
```

Comment:
But where is the implementation of LFU: I had to go search and it looks like it's inside caching_device_t as a default replacement policy.  Please add a comment explaining that to avoid confusion by future readers.